testgen/src/light_block.rs

```diff
@@ -0,0 +1,252 @@
+use gumdrop::Options;
+use serde::{Deserialize, Serialize};
+use simple_error::*;
+
+use crate::helpers::parse_as;
+use crate::validator::generate_validators;
+use crate::{Commit, Generator, Header, Validator};
+use std::str::FromStr;
+use tendermint::block::signed_header::SignedHeader;
+use tendermint::node::Id as PeerId;
+use tendermint::validator;
+use tendermint::validator::Set as ValidatorSet;
+
+/// A light block is the core data structure used by the light client.
+/// It records everything the light client needs to know about a block.
+/// NOTE: This struct & associated `impl` below are a copy of light-client's `LightBlock`.
+/// The copy is necessary here to avoid a circular dependency.
+/// Cf. https://github.com/informalsystems/tendermint-rs/issues/605
+/// TODO: fix redundant code without introducing cyclic dependency.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct TMLightBlock {
```

Comment:
It's not clear to me why do we have two types:

- `TMLightBlock` which is a copy of the same type from light-client crate
- and `LightBlock`.

The one which is necessary for testing/mocking is the former. Not sure what we use the latter for.